### PR TITLE
usb: Add USB support for nRF5340dk

### PR DESCRIPTION
**Pull request by eobalski**
This PR adds USB support to nRF5340dk alongside with samples.

### CODE REVIEW

**Comment by emob-nordic** [comment]
@koffes fyi.

File: samples/subsys/usb/audio/headphones_microphone/boards/nrf5340dk_nrf5340_cpuapp.overlay [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
```
Comment (anangl) [inline samples/subsys/usb/audio/headphones_microphone/boards/nrf5340dk_nrf5340_cpuapp.overlay:2]: 2020
Same applies to the second overlay.

File: dts/arm/nordic/nrf5340_cpuapp_peripherals.dtsi [hunk 2]
```diff
@@ -421,3 +421,16 @@ timer2: timer@11000 {
 	prescaler = <0>;
 	label = "TIMER_2";
 };
+
+usbd: usbd@36000 {
```
Comment (anangl) [inline dts/arm/nordic/nrf5340_cpuapp_peripherals.dtsi:425]: Add also a corresponding `/delete-node/` entry in `nrf5340_cpuapp_qkaa_eng_a.dtsi` (see https://github.com/zephyrproject-rtos/zephyr/commit/619a498845fa33fb830215b4caca21d8438a5d90).

@ioannisg I think in #26110 we missed that those nodes for non-functional peripherals should be deleted also in the non-secure part, i.e. `nrf5340_cpuappns_qkaa_eng_a.dtsi`. Could this be addressed in this PR or do you prefer to have it done in a separate one?

Comment (emob-nordic) [reply dts/arm/nordic/nrf5340_cpuapp_peripherals.dtsi:425]: For now I updated only secure secure part.

**Review by anangl — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Review by masz-nordic — APPROVED** [review]

**Pull request merged**